### PR TITLE
REST API Add all users to room

### PR DESCRIPTION
**Pull request by Deepakkothandan**
@RocketChat/core 

Rest API method to add all existing users to a Channel

### CODE REVIEW

**Comment by Deepakkothandan** [comment]
@geekgonecrazy This PR adds a method which provides the functionality to add all existing users to a newly create Channel. At present, it is not possible to add all existing users to a newly created channel. Please review the PR and provide your suggestions

**Comment by geekgonecrazy** [comment]
@Deepakkothandan thanks for contributing!  We'll take a look.

At first glance though this makes me very nervous.... On our demo server we have 60k+ users.  If we add all of those to a channel... That could potentially be a very costly operation.

**Comment by Deepakkothandan** [comment]
@geekgonecrazy Yeah I understand, that with a huge amount of users this could be expensive and that's the reason why only admins are allowed to do this. There are occasions, where one need to create a new channel with all existing user. Would be nice to have this feature

**Comment by Sing-Li** [comment]
@Deepakkothandan  I suggest adding an alert to (EVEN) the admin (as we have many admin who are beginning users) if the number of users is greater than a threshold (perhaps default to 100, but settings configurable).  Something like "Are you sure you want to transfer 64,509 users into this room?  Cancel , Yes"

**Comment by Deepakkothandan** [comment]
@Sing-Li Thank you for your suggestions, Added a condition if the count exceeds 500, then the call would return a error. I believe its a reasonable amount. Please provide your thoughts.

File: server/methods/addAllUserToRoom.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Meteor.methods({
+  addAllUserToRoom: function(rid) {
+    var now, room, users;
+    var userCount = RocketChat.models.Users.find().count();
+    if(userCount > 500) {
```
Comment (rodrigok) [inline server/methods/addAllUserToRoom.js]: Codacy Issue found: [Keyword "if" must be followed by whitespace.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2777659474/issues/source?bid=3199634&fileBranchId=3362492#l5)

File: server/methods/addAllUserToRoom.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      throw new Meteor.Error('error-user-limit-exceeded', 'User Limit Exceeded',{
```
Comment (rodrigok) [inline server/methods/addAllUserToRoom.js]: Codacy Issue found: [A space is required after ','.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2777659474/issues/source?bid=3199634&fileBranchId=3362492#l6)

File: server/methods/addAllUserToRoom.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+Meteor.methods({
+  addAllUserToRoom: function(rid) {
+    var now, room, users;
+    var userCount = RocketChat.models.Users.find().count();
+    if (userCount > 500) {
+      throw new Meteor.Error('error-user-limit-exceeded', 'User Limit Exceeded',{
```
Comment (rodrigok) [inline server/methods/addAllUserToRoom.js]: Codacy Issue found: [A space is required after ','.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2777811310/issues/source?bid=3199634&fileBranchId=3362492#l6)

**Comment by engelgabriel** [comment]
@RocketChat/core is this ready to merge?

**Comment by Sing-Li** [comment]
@Deepakkothandan  really thank you for your contributions.   

LGTM

**Comment by geekgonecrazy** [comment]
@Deepakkothandan thanks for making those changes, and thanks again for the contribution.  :smile:

LGTM

**Comment by sampaiodiego** [comment]
So, I see two problems here:

- The file `server/methods/addAllUserToRoom.js` is using two spaces for indentation instead of tabs. Please run `eslint .`, it should return no errors.. 
- The admin validation is done only on the REST endpoint, but the method is **unprotected**.

@Deepakkothandan can you please fix the issues above? Thank you again for the contribution =)

**Comment by geekgonecrazy** [comment]
Ah!  I missed that.  Yes for sure if you could address those.   The method needs validation on as well, as it could potentially be called from the client by unauthorized user otherwise

**Comment by Deepakkothandan** [comment]
Thank you all for reviewing the PR. @sampaiodiego Updated the PR as per your suggestions.

**Comment by sampaiodiego** [comment]
thanks @Deepakkothandan 

Maybe I wasn't clear enough, sorry about that :disappointed: 

But on the [REST route](https://github.com/RocketChat/Rocket.Chat/pull/3569/files#diff-a8ae0b76bf7b4b82a70573797676e0bdR117) you have a validation to see if the user is an admin.. this validation **have to be done** in the [Meteor method](https://github.com/RocketChat/Rocket.Chat/pull/3569/files#diff-4d01efadc67fbf7ebcb7ce7ad90857beR2) instead of the REST endpoint, because the method can be called directly from the client and the REST endpoint also call it.

**Comment by Deepakkothandan** [comment]
@sampaiodiego Thank you for the clarification, I made the required changes.

**Comment by skizzors** [comment]
how is this progress going? Ability to add all existing users to a channel is a great feature, (many competitors are missing) and very needed for our group!

**Pull request merged**